packages/typegpu/src/data/numeric.ts

```diff
@@ -144,7 +144,14 @@ const f32Cast = createDualImpl(
     return Math.fround(v);
   },
   // GPU implementation
-  (v) => snip(stitch`f32(${v})`, f32),
+  (arg): Snippet => {
+    if (!arg) return snip(0, f32);
+    if (typeof arg.value === 'number' || typeof arg.value === 'boolean') {
+      // Precomputing...
+      return snip(f32Cast[$internal].jsImpl(arg.value), f32);
+    }
+    return snip(stitch`f32(${arg})`, f32);
+  },
```

Comment:
Just did this for `f32` for now, but hopefully with my planned changes, this will be automagic for all dual impls in a future rework 🤞

Reply:
Yup, working automagically in https://github.com/software-mansion/TypeGPU/pull/1609